ecl/hql/hqlutil.cpp

```diff
@@ -6390,6 +6390,32 @@ void TempTableTransformer::reportWarning(WarnErrorCategory category, IHqlExpress
     errorProcessor.reportWarning(category, code, errorMsg.str(), str(where->sourcePath), where->lineno, where->column, where->position);
 }
 
+IHqlExpression *notePayloadFields(IHqlExpression *record, unsigned payloadCount)
+{
+    HqlExprArray fields;
+    unwindChildren(fields, record);
+    removeAttributes(fields);
```

Comment:
Is it safe to assume attributes are at the end - if not I need to maintain separate counts?